lib/time/time.go

```diff
@@ -0,0 +1,548 @@
+	x := time.Duration(d)
+
+	switch op {
+	case syntax.PLUS:
+		switch y := yV.(type) {
+		case Duration:
+			return Duration(x + time.Duration(y)), nil
+		case starlark.Int:
+			i, ok := y.Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			return Duration(x + time.Duration(i)), nil
+		case Time:
+			return Time(time.Time(y).Add(x)), nil
+		}
+
+	case syntax.MINUS:
+		switch y := yV.(type) {
+		case Duration:
+			return Duration(x - time.Duration(y)), nil
+		case starlark.Int:
+			i, ok := y.Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			return Duration(x - time.Duration(i)), nil
+		case Time:
+			// duration - time = duration
+			return nil, nil
+		}
+
+	case syntax.SLASH:
+		switch y := yV.(type) {
+		case Duration:
+			if int64(y) == 0 {
+				return nil, fmt.Errorf("%s division by zero", d.Type())
+			}
+			return d / y, nil
+		case starlark.Int:
+			i, ok := y.Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			if int64(i) == 0 {
+				return nil, fmt.Errorf("%s division by zero", d.Type())
+			}
+			return Duration(d / Duration(i)), nil
+		case Time:
+			// y := time.Time(yV.(Time))
+			// switch op {
+			// case syntax.PLUS:
+			// 	// duration + time = time
+			// 	return Time(y.Add(x)), nil
+			// case syntax.MINUS:
+			// 	// duration - time = duration
+			// 	return nil, nil
+			// }
+		}
+
+		// if int64(y) == 0 {
+		// 	return nil, fmt.Errorf("%s division by zero", d.Type())
+		// }
+		// return Duration(x / y), nil
+
+	case syntax.SLASHSLASH:
+		switch y := yV.(type) {
+		case Duration:
+			if int64(y) == 0 {
+				return nil, fmt.Errorf("%s division by zero", d.Type())
+			}
+			return d / y, nil
+		case starlark.Int:
+			i, ok := yV.(starlark.Int).Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			return d / Duration(i), nil
+		case Time:
+			// y := time.Time(yV.(Time))
+			// switch op {
+			// case syntax.PLUS:
+			// 	// duration + time = time
+			// 	return Time(y.Add(x)), nil
+			// case syntax.MINUS:
+			// 	// duration - time = duration
+			// 	return nil, nil
+			// }
+		}
+
+		// if int64(y) == 0 {
+		// 	return nil, fmt.Errorf("time.duration floored divison by zero")
+		// }
+		// return Duration(x / y), nil
+
+	case syntax.STAR:
+		switch y := yV.(type) {
+		case Duration:
+			return d * y, nil
+		case starlark.Int:
+			i, ok := y.Int64()
+			if !ok {
+				return nil, fmt.Errorf("int value out of range (want signed 64-bit value)")
+			}
+			return d * Duration(i), nil
+		case Time:
+			// y := time.Time(yV.(Time))
+			// switch op {
+			// case syntax.PLUS:
+			// 	// duration + time = time
+			// 	return Time(y.Add(x)), nil
+			// case syntax.MINUS:
+			// 	// duration - time = duration
+			// 	return nil, nil
+			// }
+		}
```

Comment:
Again I think you can only operate `duration + time` everything else makes not really sense...

Reply:
Agreed.